addons/portal/controllers/portal.py

```diff
@@ -284,14 +284,15 @@ def _check_billing_address(self, partner_sudo):
         )
         return all(partner_sudo.read(mandatory_billing_fields)[0].values())
 
-    def _get_mandatory_billing_address_fields(self, country_sudo, **kwargs):
+    def _get_mandatory_billing_address_fields(self, country_sudo, skip_address_required_fields=False, **_kwargs):
         """ Return the set of mandatory billing field names.
 
         :param res.country country_sudo: The country to use to build the set of mandatory fields.
+        :param bool skip_address_required_fields: Whether we want skip required fields check or not.
         :return: The set of mandatory billing field names.
         :rtype: set
         """
-        return self._get_mandatory_address_fields(country_sudo, **kwargs)
+        return not skip_address_required_fields and self._get_mandatory_address_fields(country_sudo) or set()
```

Comment:
too long for one line :eyes:
make an early return with an `if`, it'll be clearer